### PR TITLE
Add bh1750 driver

### PR DESCRIPTION
**Pull request by whatisbyandby**
@brentru

This PR adds the I2C Component driver for the BH1750 sensor.

Product URL: https://www.adafruit.com/product/4681
Link to Component Definition issue: [https://github.com/adafruit/Wippersnapper_Components/issues/19](https://github.com/adafruit/Wippersnapper_Components/issues/19)

This drivers uses the hp_BH1750 library to communicate with the sensor

Library Repository: https://github.com/Starmbi/hp_BH1750

This implementation uses the One Time Measurement feature of the BH1750, with the default quality and MTreg settings recommended in the datasheet for the BH1750. Each reading should take approximately 120ms.

### CODE REVIEW

File: library.properties [hunk 1]
```diff
@@ -7,4 +7,4 @@ paragraph=Arduino client for Adafruit.io WipperSnapper
...
 architectures=*
-depends=Adafruit NeoPixel, Adafruit SPIFlash, ArduinoJson, Adafruit DotStar, Adafruit SleepyDog Library, Adafruit TinyUSB Library, Adafruit AHTX0, Adafruit BME280 Library, Adafruit DPS310, Adafruit SCD30, Sensirion I2C SCD4x, Sensirion I2C SEN5X, arduino-sht, Adafruit Si7021 Library, Adafruit MQTT Library, Adafruit MCP9808 Library, Adafruit MCP9600 Library, Adafruit TSL2591 Library, Adafruit_VL53L0X, Adafruit PM25 AQI Sensor, Adafruit VEML7700 Library, Adafruit LC709203F, Adafruit seesaw Library, Adafruit BME680 Library, Adafruit MAX1704X, Adafruit ADT7410 Library
+depends=Adafruit NeoPixel, Adafruit SPIFlash, ArduinoJson, Adafruit DotStar, Adafruit SleepyDog Library, Adafruit TinyUSB Library, Adafruit AHTX0, Adafruit BME280 Library, Adafruit DPS310, Adafruit SCD30, Sensirion I2C SCD4x, Sensirion I2C SEN5X, arduino-sht, Adafruit Si7021 Library, Adafruit MQTT Library, Adafruit MCP9808 Library, Adafruit MCP9600 Library, Adafruit TSL2591 Library, Adafruit_VL53L0X, Adafruit PM25 AQI Sensor, Adafruit VEML7700 Library, Adafruit LC709203F, Adafruit seesaw Library, Adafruit BME680 Library, Adafruit MAX1704X, Adafruit ADT7410 Library, hp_BH1750
```
Comment (brentru) [inline library.properties:10]: Where is the `hp_BH1750` library coming from? Do you have a URL?

**Review by brentru — CHANGES_REQUESTED** [review]

File: library.properties (hunk 1 above)
Comment (whatisbyandby) [reply library.properties:10]: Ah, It's this library here. https://github.com/Starmbi/hp_BH1750/tags. I didn't quite understand how the library was going to get installed. Where to I need to add the URL?

Comment (whatisbyandby) [reply library.properties:10]: I was testing with the latest version 1.0.2 [https://reference.arduino.cc/reference/en/libraries/hp_bh1750/](https://reference.arduino.cc/reference/en/libraries/hp_bh1750/). Is that what you need from me?

Comment (brentru) [reply library.properties:10]: You don't need to, the library name from https://github.com/Starmbi/hp_BH1750/blob/master/library.properties#L1 is sufficient.

I just wanted to know where this was referenced.

**Review by brentru — APPROVED** [review]
This PR looks OK to me and addresses adding the BH1750 component per example code: https://github.com/Starmbi/hp_BH1750/blob/master/examples/Non_blocking/Non_blocking.ino.

**Pull request merged**